p4-16/spec/P4-16-spec.mdk

```diff
@@ -6475,6 +6514,25 @@ to two controllable entities, which is illegal.
 The `@atomic` annotation, described in Section [#sec-concurrency]
 can be used to enforce the atomic execution of a code block.
 
+## Package Documentation Annotations { #sec-package-anno }
+Package-level metadata can be declared by annotating the `main` function. Multiple k-v pairs
+can be assigned. These can be extracted by the compiler into P4Info and used for package
+management. An example is given below. Key names are not controlled by the P4 Language specification.
+The P4Runtime working group will establish some conventions.
+~ Begin P4Example
+@pkginfo(
```

Comment:
I think you would also need to change the grammar for annotations, here and in grammar.mdk.

Reply:
I'll look into that, thanks.

Reply:
OK, I enhanced the grammar and squash-merged the changes.